Flaky test: `NStepRNN`

https://jenkins.preferred.jp/job/chainer/job/chainer_pr/1962/TEST=CHAINERX_chainerx-py3,label=mn1-p100/console

```python
16:00:52 _ test_NStepRNN_param_22_{n_layers=4, hidden_size=5, input_size=4, batches=(4, 3, 2), activation='relu', in_dtypes=('float32',),  out_dtype=()}[native:0] _
16:00:52 
16:00:52 device = native:0, args = (), kwargs = {}
16:00:52 backend_config = <BackendConfig use_chainerx=True chainerx_device='native:0' use_cuda=False cuda_device=None use_cudnn='never' cudnn_deterministic=False autotune=False cudnn_fast_batch_normalization=False use_ideep='never'>
16:00:52 obj = <chainer.testing._bundle.TestNStepRNN_param_22_{n_layers=4, hidden_size=5, input_size=4, batches=(4, 3, 2), activation='relu', in_dtypes=('float32',),  out_dtype=()} object at 0x7f1cf4f58a58>
16:00:52 
16:00:52     @pytest.mark.parametrize_device(devices)
16:00:52     def entry_func(device, *args, **kwargs):
16:00:52         backend_config = _make_backend_config(device.name)
16:00:52     
16:00:52         # Forward test
16:00:52         obj = cls()
16:00:52         try:
16:00:52             obj.setup(*args, **kwargs)
16:00:52             obj.run_test_forward(backend_config)
16:00:52         finally:
16:00:52             obj.teardown()
16:00:52     
16:00:52         # If this is a NumpyOpTest instance, skip backward/double-backward
16:00:52         # tests if the forward test succeeds with acceptable errors.
16:00:52         if isinstance(obj, NumpyOpTest):
16:00:52             if obj.is_forward_successful_with_accept_errors:
16:00:52                 return  # success with expected errors
16:00:52     
16:00:52         # Backward test
16:00:52         obj = cls()
16:00:52         try:
16:00:52             obj.setup(*args, **kwargs)
16:00:52 >           obj.run_test_backward(backend_config)
16:00:52 
16:00:52 args       = ()
16:00:52 backend_config = <BackendConfig use_chainerx=True chainerx_device='native:0' use_cuda=False cuda_device=None use_cudnn='never' cudnn_deterministic=False autotune=False cudnn_fast_batch_normalization=False use_ideep='never'>
16:00:52 cls        = <class 'chainer.testing._bundle.TestNStepRNN_param_22_{n_layers=4, hidden_size=5, input_size=4, batches=(4, 3, 2), activation='relu', in_dtypes=('float32',),  out_dtype=()}'>
16:00:52 device     = native:0
16:00:52 kwargs     = {}
16:00:52 obj        = <chainer.testing._bundle.TestNStepRNN_param_22_{n_layers=4, hidden_size=5, input_size=4, batches=(4, 3, 2), activation='relu', in_dtypes=('float32',),  out_dtype=()} object at 0x7f1cf4f58a58>
16:00:52 
16:00:52 /repo/tests/chainerx_tests/op_utils.py:366: 
16:00:52 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
16:00:52 /repo/tests/chainerx_tests/op_utils.py:95: in run_test_backward
16:00:52     super(OpTest, self).run_test_backward(backend_config)
16:00:52 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:249: in run_test_backward
16:00:52     do_check()
16:00:52 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:244: in do_check
16:00:52     **self.check_backward_options)
16:00:52 /workspace/conda/envs/testenv/lib/python3.6/contextlib.py:99: in __exit__
16:00:52     self.gen.throw(type, value, traceback)
16:00:52 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:36: in raise_if_fail
16:00:52     cls.fail(message, e)
16:00:52 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:27: in fail
16:00:52     utils._raise_from(cls, message, exc)
16:00:52 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/utils/__init__.py:106: in _raise_from
16:00:52     six.raise_from(new_exc.with_traceback(orig_exc.__traceback__), None)
16:00:52 <string>:3: in raise_from
16:00:52     ???
16:00:52 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:34: in raise_if_fail
16:00:52     yield
16:00:52 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:244: in do_check
16:00:52     **self.check_backward_options)
16:00:52 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:900: in check_backward
16:00:52     detect_nondifferentiable, is_immutable_params=False
16:00:52 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:464: in run
16:00:52     self._run()
16:00:52 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:507: in _run
16:00:52     self._compare_gradients(gx_numeric, gx_backward, directions)
16:00:52 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
16:00:52 
16:00:52 self = <chainer.gradient_check._CheckBackward object at 0x7f1cf4f08dd8>
16:00:52 gx_numeric = array(2.66319670, shape=(), dtype=float64, device='native:0')
16:00:52 gx_backward = array(2.71159663, shape=(), dtype=float64, device='native:0')
16:00:52 directions = [array([[[0.01076742, 0.05109883, 0.05368832, -0.013449  , -0.00059475],
16:00:52         [0.00751749, 0.11644276, -0.02771728,... [-0.00172541, 0.03043705, 0.02962490, 0.02610405, -0.05872268]], shape=(5, 5), dtype=float64, device='native:0'), ...]
16:00:52 
16:00:52     def _compare_gradients(self, gx_numeric, gx_backward, directions):
16:00:52         atol = self.atol
16:00:52         rtol = self.rtol
16:00:52         # Compare the gradients
16:00:52         try:
16:00:52             testing.assert_allclose(
16:00:52                 gx_numeric, gx_backward, atol=atol, rtol=rtol)
16:00:52         except AssertionError as e:
16:00:52             eps = self.eps
16:00:52             xs = self.xs
16:00:52             gys = self.gys
16:00:52             f = six.StringIO()
16:00:52             f.write('check_backward failed (eps={} atol={} rtol={})\n'.format(
16:00:52                 eps, atol, rtol))
16:00:52             for i, x in enumerate(xs):
16:00:52                 f.write('inputs[{}]:\n'.format(i))
16:00:52                 f.write('{}\n'.format(x))
16:00:52             for i, gy in enumerate(gys):
16:00:52                 f.write('grad_outputs[{}]:\n'.format(i))
16:00:52                 f.write('{}\n'.format(gy))
16:00:52             for i, d in enumerate(directions):
16:00:52                 f.write('directions[{}]:\n'.format(i))
16:00:52                 f.write('{}\n'.format(d))
16:00:52             f.write('gradients (numeric):  {}\n'.format(gx_numeric))
16:00:52             f.write('gradients (backward): {}\n'.format(gx_backward))
16:00:52             f.write('\n')
16:00:52             f.write('x: numeric gradient, y: backward gradient')
16:00:52             f.write(str(e))
16:00:52 >           raise AssertionError(f.getvalue())
16:00:52 E           chainer.testing.function_link.FunctionTestError: backward is not implemented correctly
16:00:52 E           
16:00:52 E           (caused by)
16:00:52 E           AssertionError: check_backward failed (eps=0.001 atol=0.01 rtol=0.01)
16:00:52 E           inputs[0]:
16:00:52 E           array([[[0.8494544 , 0.30583617, -0.27716383, 0.83305722, -0.57502306],
16:00:52 E                   [0.02585624, -0.71387273, -0.88522607, 0.1384209 , -0.7244426 ],
16:00:52 E                   [0.43344599, 0.90352923, 0.70451033, 0.50510520, 0.41420373],
16:00:52 E                   [0.61357909, -0.98031932, 0.60162157, -0.35425758, -0.01447745]],
16:00:52 E           
16:00:52 E                  [[-0.3310697 , -0.25403684, 0.30406484, 0.28055492, 0.20148563],
16:00:52 E                   [-0.33731949, -0.97553635, 0.6830706 , 0.87297076, 0.39943492],
16:00:52 E                   [0.19549143, 0.37121272, -0.06609573, 0.90911758, 0.81600791],
16:00:52 E                   [0.64929634, 0.81536150, -0.03123625, -0.06821302, -0.57845104]],
16:00:52 E           
16:00:52 E                  [[0.75297129, -0.57692134, 0.8550387 , -0.7795847 , -0.37667289],
16:00:52 E                   [0.42011356, 0.54748666, -0.17929776, 0.64859605, -0.91732222],
16:00:52 E                   [-0.71874899, -0.39961836, 0.94632834, 0.20342836, 0.78554964],
16:00:52 E                   [-0.994879  , 0.69536465, -0.3403621 , -0.17949171, 0.03656521]],
16:00:52 E           
16:00:52 E                  [[0.40891829, 0.14154075, 0.64743268, 0.47673717, 0.86967188],
16:00:52 E                   [0.66910672, -0.39092642, -0.05486294, -0.77533025, -0.59138232],
16:00:52 E                   [-0.97095084, 0.05722094, -0.23619284, 0.13738412, 0.70809209],
16:00:52 E                   [0.72003466, 0.3741357 , 0.55455202, -0.11056276, -0.24671072]]], shape=(4, 4, 5), dtype=float32, device='native:0')
16:00:52 E           inputs[1]:
16:00:52 E           array([[-0.57541358, -0.36529768, -0.89094722, 0.38740769],
16:00:52 E                  [-0.95631987, 0.3200806 , 0.21762912, -0.06029949],
16:00:52 E                  [0.46483549, -0.47370645, 0.0191087 , -0.93938076],
16:00:52 E                  [0.96793336, -0.59378439, 0.26240245, 0.5848788 ]], shape=(4, 4), dtype=float32, device='native:0')
16:00:52 E           inputs[2]:
16:00:52 E           array([[-0.82030141, -0.57396579, -0.94964647, -0.06844151],
16:00:52 E                  [0.57449609, -0.20415585, -0.88619232, 0.86498922],
16:00:52 E                  [0.08385145, 0.80651212, 0.20115459, -0.15144402]], shape=(3, 4), dtype=float32, device='native:0')
16:00:52 E           inputs[3]:
16:00:52 E           array([[-0.85100269, -0.44231579, -0.76117045, 0.00250889],
16:00:52 E                  [-0.84267032, -0.12042955, -0.0419063 , 0.02128778]], shape=(2, 4), dtype=float32, device='native:0')
16:00:52 E           inputs[4]:
16:00:52 E           array([[0.52963668, 0.77162313, -0.58108586, 0.10729533],
16:00:52 E                  [-0.69902992, 0.27070415, -0.51095229, -0.09866288],
16:00:52 E                  [0.34294859, 0.89335662, 0.05347642, -0.74437720],
16:00:52 E                  [-0.53375590, 0.2274197 , -0.94488394, -0.72608435],
16:00:52 E                  [0.57618988, 0.08432762, -0.96574956, 0.7989344 ]], shape=(5, 4), dtype=float32, device='native:0')
16:00:52 E           inputs[5]:
16:00:52 E           array([[0.03136565, 0.47866803, -0.62007427, 0.7132458 , -0.40917152],
16:00:52 E                  [-0.87239623, 0.91819143, 0.13137795, 0.73370373, 0.44611087],
16:00:52 E                  [-0.91321146, 0.07016988, -0.39032826, 0.94034529, -0.34095755],
16:00:52 E                  [-0.85855013, -0.38011262, 0.53974783, -0.03765089, 0.9903903 ],
16:00:52 E                  [-0.94434571, 0.70945674, -0.28587639, 0.94897121, 0.9281302 ]], shape=(5, 5), dtype=float32, device='native:0')
16:00:52 E           inputs[6]:
16:00:52 E           array([-0.77758539, -0.59609723, -0.32331207, 0.22208507, -0.13733549], shape=(5,), dtype=float32, device='native:0')
16:00:52 E           inputs[7]:
16:00:52 E           array([0.43980297, -0.40853968, -0.12125328, 0.21962909, -0.73123997], shape=(5,), dtype=float32, device='native:0')
16:00:52 E           inputs[8]:
16:00:52 E           array([[0.88157189, 0.80004710, 0.64900726, 0.50991857, -0.86279649],
16:00:52 E                  [-0.83330804, 0.43215767, -0.52849042, -0.41832152, 0.54253769],
16:00:52 E                  [-0.72003216, 0.40300319, 0.9551118 , 0.44589424, 0.44867370],
16:00:52 E                  [0.45889646, -0.72593832, 0.57763833, -0.52408975, -0.11982846],
16:00:52 E                  [-0.94138873, -0.94575399, 0.20656657, -0.47664773, 0.39966249]], shape=(5, 5), dtype=float32, device='native:0')
16:00:52 E           inputs[9]:
16:00:52 E           array([[-0.84652501, 0.34884796, 0.21057701, -0.10071848, -0.82019776],
16:00:52 E                  [-0.32939616, 0.11919452, -0.63119549, 0.15758567, -0.38382626],
16:00:52 E                  [-0.42304745, -0.63078725, 0.94819969, -0.62157744, 0.69161010],
16:00:52 E                  [0.80038702, 0.06226802, -0.72637337, 0.75747639, -0.9481703 ],
16:00:52 E                  [0.81750625, 0.64312136, -0.8782795 , -0.12594484, 0.30122396]], shape=(5, 5), dtype=float32, device='native:0')
16:00:52 E           inputs[10]:
16:00:52 E           array([-0.35464364, -0.56276643, -0.87077653, 0.3689022 , 0.83285707], shape=(5,), dtype=float32, device='native:0')
16:00:52 E           inputs[11]:
16:00:52 E           array([-0.68806767, 0.84716547, 0.89005309, -0.63323480, 0.98231065], shape=(5,), dtype=float32, device='native:0')
16:00:52 E           inputs[12]:
16:00:52 E           array([[-0.06188364, 0.23179810, -0.48471180, -0.78144836, 0.59839153],
16:00:52 E                  [0.26816028, -0.47650865, 0.86177927, -0.24118295, 0.71838236],
16:00:52 E                  [0.9534738 , 0.36628917, -0.28426909, 0.97106802, -0.12793538],
16:00:52 E                  [0.90213418, -0.50878388, 0.01818117, 0.56699502, -0.37224025],
16:00:52 E                  [-0.21838   , -0.46007326, -0.32991657, 0.85594392, -0.93186235]], shape=(5, 5), dtype=float32, device='native:0')
16:00:52 E           inputs[13]:
16:00:52 E           array([[-0.19561230, 0.12731105, -0.87731826, 0.13715072, 0.42510572],
16:00:52 E                  [0.30853823, 0.48046964, -0.11197636, -0.16732799, -0.15878704],
16:00:52 E                  [-0.67381012, -0.14080568, 0.03242094, 0.5852226 , 0.54127115],
16:00:52 E                  [0.59711868, 0.23680304, 0.48170164, 0.7517299 , 0.77659529],
16:00:52 E                  [0.96262306, 0.33637053, 0.20228674, 0.35683727, -0.64080405]], shape=(5, 5), dtype=float32, device='native:0')
16:00:52 E           inputs[14]:
16:00:52 E           array([0.91621155, 0.98861188, 0.19403733, -0.07170964, 0.66751182], shape=(5,), dtype=float32, device='native:0')
16:00:52 E           inputs[15]:
16:00:52 E           array([-0.9170612 , -0.38655746, 0.38921475, 0.07906377, -0.98253006], shape=(5,), dtype=float32, device='native:0')
16:00:52 E           inputs[16]:
16:00:52 E           array([[0.38579893, 0.52563202, 0.79558039, 0.69054949, 0.41868684],
16:00:52 E                  [-0.42014262, 0.96953017, 0.77689409, -0.04033617, -0.40805402],
16:00:52 E                  [0.98161107, 0.82458168, 0.14743139, 0.60757113, -0.99974424],
16:00:52 E                  [-0.37762085, 0.52150959, -0.84577155, -0.95124465, -0.20697717],
16:00:52 E                  [0.91947389, -0.7351824 , 0.58148962, 0.42182082, 0.07090788]], shape=(5, 5), dtype=float32, device='native:0')
16:00:52 E           inputs[17]:
16:00:52 E           array([[-0.90429097, -0.92054725, -0.09558851, 0.35137   , -0.67122465],
16:00:52 E                  [-0.29578659, -0.00257726, -0.92334777, 0.82462770, 0.69109726],
16:00:52 E                  [0.50601488, -0.90792328, 0.71776646, 0.06388024, 0.79878747],
16:00:52 E                  [-0.89235115, 0.42782652, -0.88608408, -0.195221  , -0.29716352],
16:00:52 E                  [0.67298746, 0.63023466, 0.25119373, -0.74665469, 0.41208625]], shape=(5, 5), dtype=float32, device='native:0')
16:00:52 E           inputs[18]:
16:00:52 E           array([-0.89774829, 0.59127742, 0.88563168, -0.72398454, -0.63644862], shape=(5,), dtype=float32, device='native:0')
16:00:52 E           inputs[19]:
16:00:52 E           array([0.10269287, -0.36822751, -0.79856104, 0.70608675, -0.08712249], shape=(5,), dtype=float32, device='native:0')
16:00:52 E           grad_outputs[0]:
16:00:52 E           array([[[-0.23735815, 0.52001363, -0.94255716, 0.65919006, 0.32028082],
16:00:52 E                   [-0.6209473 , 0.18917365, 0.46377873, 0.88751233, -0.49165300],
16:00:52 E                   [0.2314966 , 0.24154432, 0.58149451, -0.04414974, -0.25720045],
16:00:52 E                   [0.60263324, 0.0063703 , 0.50153649, 0.3429602 , -0.05448938]],
16:00:52 E           
16:00:52 E                  [[0.52464801, -0.90347213, 0.46004403, 0.15575919, 0.85043079],
16:00:52 E                   [0.96014166, 0.79042357, 0.18119803, 0.96906197, -0.52341932],
16:00:52 E                   [-0.76275116, 0.24522324, -0.43359628, -0.95722306, -0.2419542 ],
16:00:52 E                   [-0.11494853, -0.28341243, 0.88688558, -0.12464125, -0.08190067]],
16:00:52 E           
16:00:52 E                  [[0.85849869, -0.81969553, -0.40549713, 0.41782063, 0.08608742],
16:00:52 E                   [-0.38990545, -0.40331694, 0.38840225, -0.30660897, 0.66070229],
16:00:52 E                   [0.34084687, 0.9050166 , -0.64141828, -0.41745940, -0.30947077],
16:00:52 E                   [0.2462504 , -0.07343172, 0.81560379, 0.38139763, -0.22322103]],
16:00:52 E           
16:00:52 E                  [[0.97745258, -0.07198215, 0.74151927, 0.69917578, -0.1724303 ],
16:00:52 E                   [-0.40063933, -0.79899544, 0.01059578, 0.11282919, -0.08563066],
16:00:52 E                   [0.86153352, -0.61791468, 0.83243763, -0.80794609, -0.4042598 ],
16:00:52 E                   [-0.14529167, -0.32145450, 0.33829442, 0.32699314, -0.01132289]]], shape=(4, 4, 5), dtype=float32, device='native:0')
16:00:52 E           grad_outputs[1]:
16:00:52 E           array([[0.90494418, -0.95804685, 0.62479866, 0.02006832, 0.35034886],
16:00:52 E                  [0.8176698 , -0.5659011 , -0.12208342, 0.59035689, 0.62266296],
16:00:52 E                  [0.01124540, -0.32666239, 0.82726949, -0.59996474, 0.02419488],
16:00:52 E                  [-0.72579151, 0.05314366, 0.64489096, -0.79173863, 0.8868196 ]], shape=(4, 5), dtype=float32, device='native:0')
16:00:52 E           grad_outputs[2]:
16:00:52 E           array([[0.51436478, -0.46307909, 0.94610077, -0.62350023, 0.43530646],
16:00:52 E                  [0.07920862, -0.44459248, 0.60258007, 0.93589973, -0.78741229],
16:00:52 E                  [-0.660892  , 0.48412144, -0.4786534 , -0.3782365 , 0.42485020]], shape=(3, 5), dtype=float32, device='native:0')
16:00:52 E           grad_outputs[3]:
16:00:52 E           array([[0.91380054, -0.2493941 , -0.21364133, -0.02285723, -0.30217108],
16:00:52 E                  [0.06148217, 0.70797026, 0.61465526, -0.72580659, -0.67961752]], shape=(2, 5), dtype=float32, device='native:0')
16:00:52 E           directions[0]:
16:00:52 E           array([[[0.01076742, 0.05109883, 0.05368832, -0.013449  , -0.00059475],
16:00:52 E                   [0.00751749, 0.11644276, -0.02771728, -0.0252776 , -0.03274786],
16:00:52 E                   [-0.01895078, 0.08422315, -0.11389952, -0.02251794, -0.05820616],
16:00:52 E                   [0.04475853, 0.03022555, -0.01286849, 0.06026884, -0.09882268]],
16:00:52 E           
16:00:52 E                  [[-0.02189515, -0.05728088, 0.00313735, 0.00839313, 0.00223634],
16:00:52 E                   [0.02089559, -0.0212918 , -0.07626951, 0.00508246, 0.07714276],
16:00:52 E                   [-0.03362160, 0.00723403, -0.04002116, -0.05778   , -0.05007147],
16:00:52 E                   [0.0180955 , 0.06007968, -0.06285370, -0.00728657, 0.00988877]],
16:00:52 E           
16:00:52 E                  [[-0.00742026, -0.00895986, 0.07723440, 0.05833387, 0.05208675],
16:00:52 E                   [0.03337954, -0.09770792, -0.00026069, 0.00445315, -0.04860726],
16:00:52 E                   [-0.05031715, -0.06006987, -0.00447361, 0.0069738 , 0.01409768],
16:00:52 E                   [0.06494553, 0.04056186, 0.02626249, -0.00101209, 0.04392981]],
16:00:52 E           
16:00:52 E                  [[0.01087192, -0.07563706, -0.0536923 , -0.05598474, -0.01085118],
16:00:52 E                   [-0.00708029, 0.00150846, 0.03239833, -0.03977356, -0.0147113 ],
16:00:52 E                   [0.11334439, -0.02165004, 0.03912723, -0.0060631 , 0.06520967],
16:00:52 E                   [-0.05894093, 0.0677362 , -0.01539721, -0.0435051 , -0.06982765]]], shape=(4, 4, 5), dtype=float64, device='native:0')
16:00:52 E           directions[1]:
16:00:52 E           array([[0.01865347, 0.03679084, 0.04282276, -0.04364769],
16:00:52 E                  [-0.03784344, -0.00078554, -0.05200469, 0.02608202],
16:00:52 E                  [-0.02238073, -0.01130195, 0.03946355, -0.00632144],
16:00:52 E                  [-0.00162849, 0.07971830, -0.06796869, -0.08211959]], shape=(4, 4), dtype=float64, device='native:0')
16:00:52 E           directions[2]:
16:00:52 E           array([[-0.01680512, -0.01414844, 0.02070869, 0.04120086],
16:00:52 E                  [-0.03764279, -0.06762344, 0.02159186, -0.08021014],
16:00:52 E                  [-0.00995558, 0.08150032, -0.07507283, 0.02833483]], shape=(3, 4), dtype=float64, device='native:0')
16:00:52 E           directions[3]:
16:00:52 E           array([[-0.07977369, -0.05814707, -0.13217584, 0.0577555 ],
16:00:52 E                  [-0.0284397 , 0.00015721, -0.0201043 , 0.05681496]], shape=(2, 4), dtype=float64, device='native:0')
16:00:52 E           directions[4]:
16:00:52 E           array([[0.07233252, 0.04136213, 0.00543826, -0.02327252],
16:00:52 E                  [0.04623168, -0.1475535 , -0.01345348, 0.07703370],
16:00:52 E                  [-0.05966946, 0.02980396, -0.07115996, 0.01107297],
16:00:52 E                  [0.0624229 , 0.00363886, 0.02045371, 0.00907927],
16:00:52 E                  [-0.02268835, -0.09691953, 0.03741805, -0.0557334 ]], shape=(5, 4), dtype=float64, device='native:0')
16:00:52 E           directions[5]:
16:00:52 E           array([[0.00068803, 0.01025814, -0.00110955, -0.06151666, 0.09018279],
16:00:52 E                  [0.05603790, 0.05635170, -0.0574988 , 0.02831394, -0.02653425],
16:00:52 E                  [-0.12911248, -0.04652093, -0.05526516, 0.11447563, 0.0671869 ],
16:00:52 E                  [-0.12637637, -0.06510878, 0.03394914, -0.05013529, -0.02698232],
16:00:52 E                  [-0.00172541, 0.03043705, 0.02962490, 0.02610405, -0.05872268]], shape=(5, 5), dtype=float64, device='native:0')
16:00:52 E           directions[6]:
16:00:52 E           array([0.02619918, -0.01047208, -0.0245507 , -0.11556543, 0.01290373], shape=(5,), dtype=float64, device='native:0')
16:00:52 E           directions[7]:
16:00:52 E           array([-0.02415769, -0.02238086, -0.03537638, -0.00083908, 0.0265436 ], shape=(5,), dtype=float64, device='native:0')
16:00:52 E           directions[8]:
16:00:52 E           array([[-0.01192074, -0.04068829, 0.01487278, 0.01438605, 0.01091890],
16:00:52 E                  [-0.06444975, 0.05412389, -0.05514885, -0.00544518, -0.0112519 ],
16:00:52 E                  [-0.0096175 , -0.00343456, -0.06765951, -0.06628528, 0.13064131],
16:00:52 E                  [-0.01786804, -0.06351702, -0.01166133, 0.04470843, 0.10374597],
16:00:52 E                  [-0.08231737, -0.06190091, 0.03253039, 0.03361159, -0.03968953]], shape=(5, 5), dtype=float64, device='native:0')
16:00:52 E           directions[9]:
16:00:52 E           array([[-0.02767125, -0.14183073, -0.02223835, 0.05004489, -0.06109468],
16:00:52 E                  [0.04898716, 0.02666046, 0.05341185, 0.01580660, 0.05715587],
16:00:52 E                  [-0.07707433, 0.10973204, -0.05138517, 0.03773792, -0.01250008],
16:00:52 E                  [0.00212989, -0.05681834, 0.01244173, 0.01444503, 0.08453423],
16:00:52 E                  [0.06784144, 0.04045862, -0.07549388, -0.02596685, -0.15540063]], shape=(5, 5), dtype=float64, device='native:0')
16:00:52 E           directions[10]:
16:00:52 E           array([-0.01616525, -0.03516080, -0.02221717, -0.08996415, -0.03269375], shape=(5,), dtype=float64, device='native:0')
16:00:52 E           directions[11]:
16:00:52 E           array([-0.02370922, -0.02380576, -0.03280868, 0.03731039, -0.07345456], shape=(5,), dtype=float64, device='native:0')
16:00:52 E           directions[12]:
16:00:52 E           array([[0.05753724, 0.02875274, 0.01570921, 0.02297848, 0.0459397 ],
16:00:52 E                  [0.05355636, 0.02526222, 0.01368262, 0.00273508, -0.03121884],
16:00:52 E                  [-0.05081257, 0.0341562 , -0.07545194, 0.05787796, -0.07848542],
16:00:52 E                  [0.00130955, 0.00325842, -0.00351265, -0.04763711, -0.03354196],
16:00:52 E                  [-0.02768506, 0.09758551, 0.01568708, -0.01302277, -0.06016932]], shape=(5, 5), dtype=float64, device='native:0')
16:00:52 E           directions[13]:
16:00:52 E           array([[0.05246517, 0.00179759, 0.0306477 , -0.04062660, 0.01451573],
16:00:52 E                  [0.0038126 , 0.01636324, -0.04821753, 0.0836124 , -0.03407755],
16:00:52 E                  [0.01384676, 0.00327889, -0.08034507, -0.01465255, -0.00820418],
16:00:52 E                  [0.01930835, -0.00686816, 0.06593073, 0.06945447, 0.07431205],
16:00:52 E                  [-0.07374793, -0.03125979, 0.00725098, 0.00343631, -0.05511836]], shape=(5, 5), dtype=float64, device='native:0')
16:00:52 E           directions[14]:
16:00:52 E           array([0.06598453, 0.00250449, -0.05000998, 0.11345112, -0.0111278 ], shape=(5,), dtype=float64, device='native:0')
16:00:52 E           directions[15]:
16:00:52 E           array([-0.06970016, 0.00427853, 0.03220967, -0.05324083, -0.01291783], shape=(5,), dtype=float64, device='native:0')
16:00:52 E           directions[16]:
16:00:52 E           array([[-0.11449897, -0.02903581, -0.04324389, -0.02839437, -0.0371814 ],
16:00:52 E                  [-0.00241467, -0.04704725, -0.03767155, 0.01003586, -0.04927564],
16:00:52 E                  [-0.00901805, -0.00752897, -0.04133924, 0.03185814, 0.04136824],
16:00:52 E                  [-0.0419875 , -0.06617377, -0.0272895 , -0.02227891, 0.02900399],
16:00:52 E                  [-0.05593378, -0.02831432, 0.04532194, 0.09967250, 0.0578583 ]], shape=(5, 5), dtype=float64, device='native:0')
16:00:52 E           directions[17]:
16:00:52 E           array([[0.0407318 , 0.00981076, -0.05021882, 0.04082745, -0.0529062 ],
16:00:52 E                  [-0.0363896 , 0.05569201, -0.02648398, 0.012367  , -0.02834357],
16:00:52 E                  [0.08631319, 0.02462862, 0.08955973, -0.14083296, -0.02884488],
16:00:52 E                  [-0.1679963 , 0.01431908, -0.17774855, 0.02266707, 0.04972217],
16:00:52 E                  [0.01233265, -0.11554501, 0.00077613, 0.08874318, -0.01654907]], shape=(5, 5), dtype=float64, device='native:0')
16:00:52 E           directions[18]:
16:00:52 E           array([0.05133319, -0.01647014, 0.03588119, 0.07332315, 0.00230956], shape=(5,), dtype=float64, device='native:0')
16:00:52 E           directions[19]:
16:00:52 E           array([-0.13022549, -0.04224692, 0.02487308, 0.00506163, -0.00444496], shape=(5,), dtype=float64, device='native:0')
16:00:52 E           gradients (numeric):  array(2.66319670, shape=(), dtype=float64, device='native:0')
16:00:52 E           gradients (backward): array(2.71159663, shape=(), dtype=float64, device='native:0')
16:00:52 E           
16:00:52 E           x: numeric gradient, y: backward gradient
16:00:52 E           Not equal to tolerance rtol=0.01, atol=0.01
16:00:52 E           
16:00:52 E           Mismatch: 100%
16:00:52 E           Max absolute difference: 0.04839993
16:00:52 E           Max relative difference: 0.01784924
16:00:52 E            x: array(2.663197)
16:00:52 E            y: array(2.711597)
16:00:52 E           
16:00:52 E           assert_allclose failed: 
16:00:52 E             shape: () ()
16:00:52 E             dtype: float64 float64
16:00:52 E             i: (0,)
16:00:52 E             x[i]: 2.663196697873726
16:00:52 E             y[i]: 2.7115966277450183
16:00:52 E             relative error[i]: 0.01784923663647641
16:00:52 E             absolute error[i]: 0.04839992987129227
16:00:52 E             relative tolerance * |y[i]|: 0.027115966277450183
16:00:52 E             absolute tolerance: 0.01
16:00:52 E             total tolerance: 0.03711596627745018
16:00:52 E           x: 2.6631967
16:00:52 E           y: 2.71159663
16:00:52 
16:00:52 atol       = 0.01
16:00:52 d          = array([-0.13022549, -0.04224692, 0.02487308, 0.00506163, -0.00444496], shape=(5,), dtype=float64, device='native:0')
16:00:52 directions = [array([[[0.01076742, 0.05109883, 0.05368832, -0.013449  , -0.00059475],
16:00:52         [0.00751749, 0.11644276, -0.02771728,... [-0.00172541, 0.03043705, 0.02962490, 0.02610405, -0.05872268]], shape=(5, 5), dtype=float64, device='native:0'), ...]
16:00:52 eps        = 0.001
16:00:52 f          = <_io.StringIO object at 0x7f1c51576b88>
16:00:52 gx_backward = array(2.71159663, shape=(), dtype=float64, device='native:0')
16:00:52 gx_numeric = array(2.66319670, shape=(), dtype=float64, device='native:0')
16:00:52 gy         = array([[0.91380054, -0.2493941 , -0.21364133, -0.02285723, -0.30217108],
16:00:52        [0.06148217, 0.70797026, 0.61465526, -0.72580659, -0.67961752]], shape=(2, 5), dtype=float32, device='native:0')
16:00:52 gys        = (array([[[-0.23735815, 0.52001363, -0.94255716, 0.65919006, 0.32028082],
16:00:52         [-0.6209473 , 0.18917365, 0.46377873,...      [0.06148217, 0.70797026, 0.61465526, -0.72580659, -0.67961752]], shape=(2, 5), dtype=float32, device='native:0'))
16:00:52 i          = 19
16:00:52 rtol       = 0.01
16:00:52 self       = <chainer.gradient_check._CheckBackward object at 0x7f1cf4f08dd8>
16:00:52 x          = array([0.10269287, -0.36822751, -0.79856104, 0.70608675, -0.08712249], shape=(5,), dtype=float32, device='native:0')
16:00:52 xs         = (array([[[0.8494544 , 0.30583617, -0.27716383, 0.83305722, -0.57502306],
16:00:52         [0.02585624, -0.71387273, -0.88522607... [-0.94434571, 0.70945674, -0.28587639, 0.94897121, 0.9281302 ]], shape=(5, 5), dtype=float32, device='native:0'), ...)
16:00:52 
16:00:52 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:537: FunctionTestError
```

--- a/tests/chainerx_tests/unit_tests/routines_tests/test_rnn.py
+++ b/tests/chainerx_tests/unit_tests/routines_tests/test_rnn.py
@@ -429,7 +429,9 @@ class TestNStepBiGRU(op_utils.ChainerOpTest):
 
 
 @op_utils.op_test(['native:0', 'cuda:0'])
-@op_utils.fix_random()  # ReLU activation is unstable around 0 but can seemingly not be dodged automatically.  # NOQA
+# ReLU activation is unstable around 0 but can seemingly not be dodged
+# automatically.
+@op_utils.fix_random()
 @chainer.testing.parameterize(*(
     chainer.testing.product([
         chainer.testing.from_pytest_parameterize(
@@ -535,7 +537,9 @@ class TestNStepRNN(op_utils.ChainerOpTest):
 
 
 @op_utils.op_test(['native:0', 'cuda:0'])
-@op_utils.fix_random()  # ReLU activation is unstable around 0 but can seemingly not be dodged automatically.  # NOQA
+# ReLU activation is unstable around 0 but can seemingly not be dodged
+# automatically.
+@op_utils.fix_random()
 @chainer.testing.parameterize(*(
     chainer.testing.product([
         chainer.testing.from_pytest_parameterize(

--- a/tests/chainerx_tests/unit_tests/routines_tests/test_rnn.py
+++ b/tests/chainerx_tests/unit_tests/routines_tests/test_rnn.py
@@ -429,6 +429,7 @@ class TestNStepBiGRU(op_utils.ChainerOpTest):
 
 
 @op_utils.op_test(['native:0', 'cuda:0'])
+@op_utils.fix_random()  # ReLU activation is unstable around 0 but can seemingly not be dodged automatically.  # NOQA
 @chainer.testing.parameterize(*(
     chainer.testing.product([
         chainer.testing.from_pytest_parameterize(
@@ -534,6 +535,7 @@ class TestNStepRNN(op_utils.ChainerOpTest):
 
 
 @op_utils.op_test(['native:0', 'cuda:0'])
+@op_utils.fix_random()  # ReLU activation is unstable around 0 but can seemingly not be dodged automatically.  # NOQA
 @chainer.testing.parameterize(*(
     chainer.testing.product([
         chainer.testing.from_pytest_parameterize(

